### DISCUSSION TITLE
Changing to real ads for release

### DISCUSSION THREAD

**Discussion by Felipe634** [Poing-Studios/Godot-AdMob-Android-iOS#62] [Q&A]
To change to real ads ready for release, is all I have to do tick the box saying 'is_real' in the 'MobileAds.tscn', or is there more to be done, also, does it matter if I leave it configured to 'release.a' even for testing?

**Comment by gumaciel** [comment]
Hey, thank you for opening this issue.

> To change to real ads ready for release, is all I have to do tick the box saying 'is_real' in the 'MobileAds.tscn', or is there more to be done?

When you tick the box saying `is_real` we are doing this: https://developers.google.com/admob/android/test-ads

- Google:
> It's important to enable test ads during development so that you can click on them without charging Google advertisers. 
> If you click on too many ads without being in test mode, you risk your account being flagged for invalid activity.

So we always recommend you uncheck the box `is_real` when you are testing your application, when you go to to release your app just check the box `is_real`.

Also: You will need to do this:

- Add your [AdMob App ID](https://support.google.com/admob/answer/7356431) to your app's `res://android/build/AndroidManifest.xml` file by adding a `<meta-data>` tag with name `com.google.android.gms.ads.APPLICATION_ID`, as shown below.
```xml
<!-- Sample AdMob app ID: ca-app-pub-3940256099942544~3347511713 -->
<meta-data
	tools:replace="android:value"
	android:name="com.google.android.gms.ads.APPLICATION_ID"
	android:value="ca-app-pub-xxxxxxxxxxxxxxxx~yyyyyyyyyy"/>
```

And this: (this is for banner, but can be for interstitial and rewarded) 
- https://support.google.com/admob/answer/7311346?hl=en

Please watch this video marked at this time, might help you better: https://youtu.be/AZFV0L0DXoU?t=84

> Does it matter if I leave it configured to 'release.a' even for testing?
No, you can test using the `release.a`, theres no big diferences between the `release` and `debug`.

**Reply by Felipe634** [reply]
Thank you, I am assuming that you only do the meta data part for android? Also, where do I put my unique ad units for ads, which script is it in?

**Reply by gumaciel** [reply]
> I am assuming that you only do the meta data part for android?

Yes, and the `meta data` of UMP is optional, only is required if you are releasing your app to EEA: https://ec.europa.eu/eurostat/statistics-explained/index.php/Glossary:European_Economic_Area_(EEA)

For iOS you need to do that:
![identifer](https://camo.githubusercontent.com/4a0077bdbb4f11a1518ed0dd079e50529b2e7e5780c309b1a88a09d3cd253b42/68747470733a2f2f692e696d6775722e636f6d2f3174634b5878352e706e67)
https://developers.google.com/admob/ios/quick-start#update_your_infoplist

Please be sure to do all these steps: https://github.com/Poing-Studios/Godot-AdMob-Android-iOS#ios-v33

> where do I put my unique ad units for ads, which script is it in?

You can do that on `MobileAds.tscn` or `MobileAds.gd` on `Unit Ids` part:
![mobileads](https://user-images.githubusercontent.com/42573485/112758906-49a34e80-8ff9-11eb-9133-d214ea99a5ac.png)

**Reply by Felipe634** [reply]
One more thing, is it necessary or option to request app tracking transparancy? And where I change my ad units, do I delete the default android ones.

**Reply by gumaciel** [reply]
I would say it is necessary, since from version 14 of iOS Apple is demanding this, if you do not put the App Tracking Transparency everything will be fine, but you may have less revenue because AdMob will not be able to know what's the best ad to show to the user.

**Reply by gumaciel** [reply]
Read more about here: https://stackoverflow.com/a/63522856

**Reply by gumaciel** [reply]
> where I change my ad units, do I delete the default android ones.

You need to replace the default with yours that you created here: https://admob.google.com/

Note: The default on `MobileAds.tscn` are test ads, you need to create your own ones and then replace.

**Reply by Felipe634** [reply]
So if I add my ad units for android in that section, and the ad units for IOS in that section, I can export this for android aswell without needing to create a separate project for android.

**Reply by gumaciel** [reply]
Yes, that's it.

You just need to put the Unit Ids for the respective system, then the AdMob API will do the rest for you.

**Reply by gumaciel** [reply]
Try it out with the example project, if you have a Mac you can use the Xcode Simulator to see how it works without an iPhone.

**Reply by Felipe634** [reply]
For the app tracking transparancy, is all I need to do is the bit in xcode which is in the instructions, or is there more for it to work.

**Reply by gumaciel** [reply]
Just the Xcode instructions:

![tracking](https://camo.githubusercontent.com/aa357efc763133b75220ca6a29d382efb69c975126c994b1672fc41fd147d2ef/68747470733a2f2f646576656c6f706572732e676f6f676c652e636f6d2f61646d6f622f696d616765732f696466612f656469746f722e706e67)

**Reply by gumaciel** [reply]
Oops, sorry, it's already placed when you export the game (thanks for new iOS Plugin), i forgot to delete this part on the instructions.

**Reply by Felipe634** [reply]
I submitted my app for review and it was rejected because of how I did the app tracking transparancy, it said that the permission request pop up is showing up before the permission request to encourage the user to accept tracking, how do I fix this? Also the screenshot that they gave me to change showed a picture of a pop up saying 'consent' or 'do not consent', and at the top of that pop up it says test ads even though I changed to real ads or does that not matter?

**Reply by gumaciel** [reply]
Very strange, I submitted a game with this plugin a few days ago and it was approved normally.

![Captura de Tela 2021-05-03 às 09 14 03](https://user-images.githubusercontent.com/20030153/116874572-ea37f000-abef-11eb-8fb7-92446775c7b9.png)
Those were the settings

If you have Discord, please join here: https://discord.gg/FAMNVzBJ i can help you faster

**Comment by gumaciel** [comment]
I will convert this issue to a discussion because was a question, please open a discussion if you have other doubts: https://github.com/Poing-Studios/Godot-AdMob-Android-iOS/discussions

**Reply by alexisbogarin** [reply]
how long does it take for the actual ads to appear? :)

**Reply by gumaciel** [reply]
Hi, if you are testing with Android, then will appear in few hours, if is on Apple, you need to add the app on AdMob.com
